DataFormats/TotemRawData/interface/TotemFramePosition.h

```diff
@@ -0,0 +1,112 @@
+/****************************************************************************
+*
+* This is a part of the TOTEM testbeam/monitoring software.
+* This is a part of the TOTEM offline software.
+* Authors: 
+*   Jan Kašpar (jan.kaspar@gmail.com) 
+*
+****************************************************************************/
+
+
+#ifndef _TotemFramePosition_h_
+#define _TotemFramePosition_h_
+
+#include <iostream>
+#include <string>
+
+/**
+ * Uniquely identifies the DAQ channel through which a VFAT frame has been received.
+ * 
+ * The internal representation has the following structure:
+ * \verbatim
+ * |                               32 bits raw position                                 |
+ * | 12 bits | 2 bits |  3 bits   |  5 bits   |  2 bits   | 4 bits |       4 bits       |
+ * |  empty  | empty  | SubSystem | TOTFED ID | OptoRx ID | GOH ID | index within fiber |
+ * |         |   (this part is encoded in OptoRx header)  |                             |
+ * \endverbatim
+ * According to the convention SubSystemId goes from 1 to 6, TOTFEDId from 1 to 21 and OptoRx from 1 to 3.
+ **/
+class TotemFramePosition
```

Comment:
This is essentially an electronics channel ID.
In normal circumstances these should not be in the DataFormats.
If some recabling happens, the values will change, but this will have no relevance to physics cell readouts.